### PR TITLE
chore: implements skeleton for notifications fee details

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR implements a skeleton placeholder for Network fee component on Notifications details since this information is fetched at runtime. This PR also implements a placeholder for the case where no network fee is provided whatsoever. 

Ticket: [NOTIFY-1235](https://consensyssoftware.atlassian.net/browse/NOTIFY-1235)
## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **Before**

Simulator Screenshot - iPhone 15 - 2024-10-17 at 16 04 52

### **After**
[Loading State]
Simulator Screenshot - iPhone 15 - 2024-10-17 at 16 03 51

[No Network Fee State]
Simulator Screenshot - iPhone 15 - 2024-10-17 at 16 08 04

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

[NOTIFY-1235]: https://consensyssoftware.atlassian.net/browse/NOTIFY-1235?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx [hunk 1]
```diff
@@ -38,13 +40,21 @@ type NetworkFee = Awaited<ReturnType<ModalFieldNetworkFee['getNetworkFees']>>;
...
     getNetworkFees()
-      .then((result) => setData(result))
-      .catch(() => setData(undefined));
+      .then((result) => {
+        setData(result);
+        setIsLoading(false);
+      })
+      .catch(() => {
+        setData(undefined);
+        setIsLoading(false);
```
Comment (tommasini) [inline app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx:53]: the setIsLoading false should be on finally not in the catch block right?

Comment (Jonathansoufer) [reply app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx:54]: Good catch. Fixed.

**Review by tommasini — APPROVED** [review]

File: app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx [hunk 2]
```diff
@@ -38,13 +40,22 @@ type NetworkFee = Awaited<ReturnType<ModalFieldNetworkFee['getNetworkFees']>>;
...
   const [data, setData] = useState<NetworkFee | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
   useEffect(() => {
+    setIsLoading(true);
     getNetworkFees()
-      .then((result) => setData(result))
-      .catch(() => setData(undefined));
+      .then((result) => {
+        setData(result);
+        setIsLoading(false);
```
Comment (mathieuartu) [inline app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx:49]: This one is redundant with the one in the `finally` statement

**Review by mathieuartu — COMMENTED** [review]
LGTM! Just added a small change

File: app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx (hunk 2 above)
Comment (Jonathansoufer) [reply app/components/Views/Notifications/Details/Fields/NetworkFeeField.tsx:49]: It was addressed already [here](https://github.com/MetaMask/metamask-mobile/pull/11840/commits/fed30735c209d195d1690fe47167ea123bd50e4c)

**Pull request merged**